### ISSUE TITLE
Cannot reproduce advertised accuracy numbers for mobilenet_v1-1.0

### ISSUE THREAD

**Issue by shatak2012** [neuralmagic/sparsezoo#106]
**Describe the bug**
I am not able to exactly reproduce the advertised accuracy number of 70.9 top1 accuracy 
zoo:cv/classification/mobilenet_v1-1.0/pytorch/sparseml/imagenet/base-none

Since this model came from pytorch, I am assuming it's the standard mean and standard deviation used by pytorch models which is  mean = [0.485, 0.456, 0.406] and std = [0.229, 0.224, 0.225].  Is there a different mean/stddev that I need to use for this model or some other setting?

**Expected behavior**

**Environment**
Include all relevant environment information:
1. OS [e.g. Ubuntu 18.04]: Windows
2. Python version [e.g. 3.7]: 3.6
3. SparseZoo version or commit hash [e.g. 0.1.0, `f7245c8`]:  0.6.0

**Errors**
Expected top1 accuracy is 70.9 got 68.9 only. This happens for this particular model only. With Resnet-v1-101 I can reproduce the accuracy numbers with same setttings.

**Additional context**
Add any other context about the problem here. Also include any relevant files.

**Comment by markurtz** [comment]
Hi @shatak2012, the means and std do look correct. What is the code you're using to run through tests for? Additionally, could you try and validate the accuracy for one of the pruned MobileNets so we can identify if this is an issue with that specific model or something else?

Thanks,
Mark

**Comment by shatak2012** [comment]
I tried this with the original model. 

`
import torch
import os
import sys
from torchvision import models
import torch.utils.data as data
import torchvision.transforms as transforms
from torchvision import datasets
from PIL import Image
from tqdm import tqdm
from sparseml.pytorch.models.classification.mobilenet import han_mobilenet

class stats(object):
  def __init__(self):
    self.reset()

  def reset(self):
    self.val = 0
    self.avg = 0
    self.sum = 0
    self.count = 0

  def update(self, val, n=1):
    self.val = val;
    self.sum += val * n
    self.count += n
    self.avg = self.sum / self.count


class ImagenetDataset(data.Dataset):
  def __init__(self, imglist, transform=None):
    self.imglist = imglist
    self.transform = transform
    self.images = []
    self.target = []
    print(imglist)
    
    inputfile = open(self.imglist, 'r')
    line = inputfile.readline()
    lines = inputfile.readlines()

    
    for line in lines:
      val = line.split()
      self.images.append(val[0])
      self.target.append(int(val[1]))
      
      inputfile.close()

  def __getitem__(self, index):
    img_id = self.images[index]
    target = self.target[index]    
    image = Image.open(img_id).convert('RGB')
    
    if self.transform is not None:
      image = self.transform(image)
      
    return image, target

  def __len__(self):
    return len(self.images)


def accuracy(output, target, topk=(1, 5)):
  maxk = max(topk)
  batchsize = len(target)
  scores, pred = output.topk(maxk, 1, True, True)
  pred = pred.t()
  correct = pred.eq(target.view(1, -1).expand_as(pred))
  #scores[batch][5]
  res = []
  for k in topk:
    correctk = correct[:k].float().sum()
    res.append(correctk.mul_(100.0/batchsize))
  return res

def main(argv):
  transform = transforms.Compose([transforms.Resize(256),
                                  transforms.CenterCrop(224),
                                  transforms.ToTensor(),
                                  transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]),
                                  ])

  filename = argv[1]
  modelname = argv[2]

  if os.path.isfile(filename):
    dataset = ImagenetDataset(filename, transform)
  elif os.path.isdir(filename):
    dataset = datasets.ImageFolder(filename, transform=transform)

  val_loader = torch.utils.data.DataLoader(dataset=dataset, 
                               batch_size=128,
                               shuffle=False,
                               num_workers=8)

  model = han_mobilenet(pretrained_path=modelname, load_strict=False)
  model.eval()
  pre_trained_model = torch.load(modelname)
  trained_layers = list(pre_trained_model['state_dict'].items())
  model.classifier.fc[-1].weight.data = trained_layers[-2][1]
  model.classifier.fc[-1].bias.data = trained_layers[-1][1]

  top1Val = stats()
  top5Val = stats()
  val_bar = tqdm(val_loader)

  for inputs, labels in val_bar:
    outputs = model(inputs)
    acc = accuracy(outputs[1], labels)
    top1Val.update(acc[0], inputs.size(0))
    top5Val.update(acc[1], inputs.size(0))
    val_bar.set_description('Validation Acc: {:.4f}'.format(top1Val.avg))

  print('Top-1 : ', top1Val.avg)
  print('Top-5 : ', top5Val.avg)
`

python run_mobilenet.py original_val50k.txt model.pth

Validation Acc: 68.9000: 100%|███████████████████████████████████████████████████████| 391/391 [26:26<00:00,  4.06s/it]
Top-1 :  tensor(68.9000)
Top-5 :  tensor(88.8700)

**Comment by shatak2012** [comment]
Just to clarify changing han_mobilnet to mobilnet doesn't improve the accuracy either for mobilenet_v1. 

When trying to reproduce the accuracy with mobilnet_v2 base model it works fine, however I can't find corresponding pruned mobilnetv2 model. Any reason you don't have mobilnet_v2 pruned model? Is it hard to prune?

**Comment by markurtz** [comment]
HI @shatak2012, wanted to give an update on this. It's still on our active backlog and we're planning to dive in more after our next release goes out from mid to end of January.

Thanks,
Mark

**Comment by jeanniefinks** [comment]
Hi @shatak2012 
We took some time to explore this task; in light of other items higher on our roadmap, we are no longer continuing further development of creating pruned MobileNetV2 and MobileNetV3 versions at this time. We hope there are some other alternatives in our [sparsezoo.neuramagic.com](https://sparsezoo.neuramagic.com) for consideration. If you have other questions, please don't hesitate to reach out. 

Sincerely,  Jeannie / Neural Magic
